Source/Extensions/CAShapeLayerExtension.swift

```diff
@@ -13,18 +13,14 @@ extension CAShapeLayer {
     /// - Parameter style: Protocol that provide which edge or corner will be added with style
     func createCustomBorder(with style: ElementBorderStyle) {
 
-        // if path not nil then it's already added so no need to draw lines or curves again
-        guard self.path == nil else { return }
-
```

Comment:
You cannot remove this without making change to how `createCustomBorder` is consumed. This bit of code in `BorderView` will end up stacking border over border, especially if inside a dequeable cell that keeps reusing.

![Screenshot 2022-11-29 at 09 30 59](https://user-images.githubusercontent.com/102028170/204491871-15c68e5d-5bcc-43ca-b20f-b8043a8567a9.png)

As I have previously suggested, dropping that `guard let style = style` and the subsequent line in the `layoutSubviews` should be removed.

Reply:
Cool, haven't noticed that, was under the assumption we're adding layer on top of layer. I do think though that we would only ever need to createCustomBorder once, when the style is injected, then only update the frame as subviews are told to layout?